src/main/java/backend/resource/Model.java

```diff
@@ -211,6 +211,22 @@ private void ______OPERATIONS_____() {}
                 () -> logger.error("Issue " + issueId + " not found in model for " + repoId));
     }
 
+    /**
+     * Replaces the milestone of an issue specified by {@code issueId} with {@code milestone}
+     * @param issueId
+     * @param milestone
+     * @return the modified TurboIssue if successful
+     */
+    public synchronized Optional<TurboIssue> replaceIssueMilestone(int issueId, Integer milestone) {
+        Optional<TurboIssue> issueLookUpResult = getIssueById(issueId);
+        return Utility.safeFlatMapOptional(issueLookUpResult,
```

Comment:
Agree. But what should the Logic do to handle this exception? If the milestone change is not reverted, it might cause problems with the models later on (at the moment, this seems to be the case if the label/milestone revert fails.)